Xamarin.ExposureNotification/ExposureNotification.android.cs

```diff
@@ -1,114 +1,112 @@
-using Android.Gms.ExposureNotifications;
 using Android.App;
+using Android.Gms.Nearby.ExposureNotification;
+using Nearby = Android.Gms.Nearby.NearbyClass;
 
 [assembly: UsesPermission(Android.Manifest.Permission.Bluetooth)]
 
 namespace Xamarin.ExposureNotifications
 {
 	public static partial class ExposureNotification
 	{
-		static ExposureNotificationClient instance;
+		static IExposureNotificationClient instance;
 
-		static ExposureNotificationClient Instance
-			=> instance ??= new ExposureNotificationClient();
+		static IExposureNotificationClient Instance
+			=> instance ??= Nearby.GetExposureNotificationClient(Application.Context);
 
 		static async Task PlatformStart(IExposureNotificationHandler handler)
 		{
 			var c = handler.Configuration;
 
-			// TODO: Verify mapping of config
-			await Instance.Start(new ExposureNotificationClient.ExposureConfiguration
-			{
-				AttenuationScores = c.AttenuationScores,
-				DurationScores = c.DurationScores,
-				DaysSinceLastExposureScores = c.DaysScores,
-				TransmissionRiskScores = c.TransmissionRiskScores
-			});
+			// TODO: weights are missing
+
+			var config = new ExposureConfiguration.ExposureConfigurationBuilder()
+				.SetAttenuationScores(c.AttenuationScores)
+				.SetDurationScores(c.DurationScores)
+				.SetDaysSinceLastExposureScores(c.DaysScores)
+				.SetTransmissionRiskScores(c.TransmissionRiskScores)
+				.Build();
+
+			await Instance.StartAsync(config);
 		}
 
 		static Task PlatformStop()
-			=> Instance.Stop();
+			=> Instance.StopAsync();
 
-		static Task<bool> PlatformIsEnabled()
-			=> Instance.IsEnabled();
+		static async Task<bool> PlatformIsEnabled()
+			=> await Instance.IsEnabledAsync();
 
 		// Gets the contact info of anyone the user had contact with who was diagnosed
 		static async Task<IEnumerable<ExposureInfo>> PlatformGetExposureInformation()
 		{
-			var details = await Instance.GetExposureInformation();
+			var details = await Instance.GetExposureInformationAsync();
 
 			return details.Select(d => new ExposureInfo(
 				DateTimeOffset.UnixEpoch.AddMilliseconds(d.DateMillisSinceEpoch).UtcDateTime,
 				TimeSpan.FromMinutes(d.DurationMinutes),
 				d.AttenuationValue,
 				d.TotalRiskScore,
-				(RiskLevel)((int)d.TransmissionRiskLevel)));
+				(RiskLevel)d.TransmissionRiskLevel));
 		}
 
 		// Call this when the user has confirmed diagnosis
 		static async Task PlatformSubmitSelfDiagnosis()
 		{
-			var selfKeys = await Instance.GetTemporaryExposureKeyHistory();
+			var selfKeys = await Instance.GetTemporaryExposureKeyHistoryAsync();
 
 			await Handler.UploadSelfExposureKeysToServer(
 				selfKeys.Select(k => new TemporaryExposureKey(
-					k.KeyData,
-					k.RollingStartNumber,
-					TimeSpan.FromMinutes(k.RollingDuration),
+					k.GetKeyData(),
+					k.RollingStartIntervalNumber,
+					TimeSpan.Zero, // TODO: TimeSpan.FromMinutes(k.RollingDuration),
 					(RiskLevel)k.TransmissionRiskLevel)));
```

Comment:
Should we do From/To native here?

Reply:
We could... Might be better in case Google changes. But our enum is 1:1 of theirs.